### ISSUE TITLE
Release V2 to NPM

### ISSUE THREAD

**Issue by cjdell** [johanneslumpe/react-native-fs#138]
I want to make a few minor but none the less breaking changes to the API. I welcome feedback on this. Feel free to create new issues under this milestone. When we're all happy we should publish a new V2.x.x to NPM.

**Comment by johanneslumpe** [comment]
@cjdell I completely agree! I think we shouldn't push v2 back for too long, as we already have a lot of new features in master which aren't accessible via npm. Thoughts?

**Comment by cjdell** [comment]
@johanneslumpe Will try my hardest to get V2 publishable this week.

**Comment by johanneslumpe** [comment]
@cjdell That would be amazing. I know I haven't done any work in recent times on this, but I'm so busy with other stuff in life at the moment :/ Eventually I will come back to this though ;)

**Comment by johanneslumpe** [comment]
@cjdell https://github.com/johanneslumpe/react-native-fs/issues/148 is already fixed in the laster master, right?

**Comment by cjdell** [comment]
@johanneslumpe I think so, I'll check. Going to see if I can finish off V2 this evening.

**Comment by cjdell** [comment]
@johanneslumpe We have a prerelease. I'm only one man so I wouldn't count on it being bug free just yet. Could do with some feedback from others before we push it out to NPM. :-)

https://github.com/johanneslumpe/react-native-fs/releases/tag/v2.0.1-0

**Comment by CesarLanderos** [comment]
i will be testing it on android this week and the next, thanks for the effort!, i will submit an issue if i find something wrong :)

**Comment by CesarLanderos** [comment]
just did a quick test, everything works fine!, my app does all sort of things with this package, in fact, its one of the core packages, move files, create folders, check if exists, download with progress indicator, thos are the main operations i remember.

before this version i had 2 problems:

- progress divider did not work and on emulator was slow as hell, solved!
- some weird behavior with stop download (maybe this is my fault, haven't found the bug yet)

need to get more in deep with my stop download problem, but so far everything works as expected at least in my app, thanks!

**Comment by johanneslumpe** [comment]
@cjdell How much more testing should we do on this? I won't be able to test (have no project using RN atm..)

**Comment by johanneslumpe** [comment]
@cjdell @CesarLanderos I just published a release candidate to npm: `2.0.1.-rc.1`

**Comment by CesarLanderos** [comment]
yei! :tada:, i've been using the RC since i posted that last comment and everything works great!, stop download behavior work as expected, i also opened an issue about the actual behavior.

since friday i've been fixing lots of bugs on my app that uses RNFS, and nothing was related to the package in particular, breaking changes are well explained and had no trouble making the appropriate changes in my app, as i mentioned before, i think i use most of the methods exposed by this package, so, it seems stable for me right now :smiley:

**Comment by cjdell** [comment]
New prerelease out people! https://github.com/johanneslumpe/react-native-fs/releases/tag/v2.0.1-rc.2

**Comment by johanneslumpe** [comment]
and just published to npm!
